src/chronify/time_series_checker.py

```diff
@@ -44,17 +48,60 @@ def check_timestamps(self) -> None:
 
+    @staticmethod
+    def _has_prevailing_time_zone(lst: list[tzinfo | None]) -> bool:
+        for tz in lst:
+            if is_prevailing_time_zone(tz):
+                return True
+        return False
+
     def _check_expected_timestamps(self) -> int:
+        """Check that the timestamps in the table match the expected timestamps."""
+        if isinstance(self._time_generator, DatetimeRangeGeneratorExternalTimeZone):
+            return self._check_expected_timestamps_with_external_time_zone()
+        return self._check_expected_timestamps_datetime()
+
+    def _check_expected_timestamps_datetime(self) -> int:
+        """For tz-naive or tz-aware time without external time zone column"""
         expected = self._time_generator.list_timestamps()
         time_columns = self._time_generator.list_time_columns()
         stmt = select(*(self._table.c[x] for x in time_columns)).distinct()
         for col in time_columns:
             stmt = stmt.where(self._table.c[col].is_not(None))
         df = read_database(stmt, self._conn, self._schema.time_config)
         actual = self._time_generator.list_distinct_timestamps_from_dataframe(df)
+        expected = sorted(set(expected))  # drop duplicates for tz-naive prevailing time
         check_timestamp_lists(actual, expected)
         return len(expected)
 
+    def _check_expected_timestamps_with_external_time_zone(self) -> int:
+        """For tz-naive time with external time zone column"""
+        assert isinstance(self._time_generator, DatetimeRangeGeneratorExternalTimeZone)  # for mypy
+        expected_dct = self._time_generator.list_timestamps_by_time_zone()
+        time_columns = self._time_generator.list_time_columns()
+        assert isinstance(self._schema.time_config, DatetimeRangeWithTZColumn)  # for mypy
+        time_columns.append(self._schema.time_config.get_time_zone_column())
+        stmt = select(*(self._table.c[x] for x in time_columns)).distinct()
+        for col in time_columns:
+            stmt = stmt.where(self._table.c[col].is_not(None))
+        df = read_database(stmt, self._conn, self._schema.time_config)
+        actual_dct = self._time_generator.list_distinct_timestamps_by_time_zone_from_dataframe(df)
+
+        if sorted(expected_dct.keys()) != sorted(actual_dct.keys()):
+            msg = (
+                "Time zone records do not match between expected and actual from table "
+                f"\nexpected: {sorted(expected_dct.keys())} vs. \nactual: {sorted(actual_dct.keys())}"
+            )
+            raise InvalidTable(msg)
+
+        for tz_name in expected_dct.keys():
+            # drops duplicates for tz-naive prevailing time
+            expected = sorted(set(expected_dct[tz_name]))
+            actual = actual_dct[tz_name]
+            check_timestamp_lists(actual, expected, msg_prefix=f"For {tz_name}\n")
+        # return len by preserving duplicates for tz-naive prevailing time
+        return len(expected_dct[tz_name])
```

Comment:
This looks like a bug because tz_name will be the last key in expected_dct or unbound if there are no keys.